pyqtgraph/functions.py

```diff
@@ -1678,69 +1678,52 @@ def downsample(data, n, axis=0, xvals='subsample'):
-    ## so the original (slower) method is left here for emergencies:
-        #path.moveTo(x[0], y[0])
-        #if connect == 'all':
-            #for i in range(1, y.shape[0]):
-                #path.lineTo(x[i], y[i])
-        #elif connect == 'pairs':
-            #for i in range(1, y.shape[0]):
-                #if i%2 == 0:
-                    #path.lineTo(x[i], y[i])
-                #else:
-                    #path.moveTo(x[i], y[i])
-        #elif isinstance(connect, np.ndarray):
-            #for i in range(1, y.shape[0]):
-                #if connect[i] == 1:
-                    #path.lineTo(x[i], y[i])
-                #else:
-                    #path.moveTo(x[i], y[i])
-        #else:
-            #raise Exception('connect argument must be "all", "pairs", or array')
-
-    ## Speed this up using >> operator
-    ## Format is:
-    ##    numVerts(i4)
-    ##    0(i4)   x(f8)   y(f8)    <-- 0 means this vertex does not connect
-    ##    1(i4)   x(f8)   y(f8)    <-- 1 means this vertex connects to the previous vertex
-    ##    ...
-    ##    cStart(i4)   fillRule(i4)
-    ##
-    ## see: https://github.com/qt/qtbase/blob/dev/src/gui/painting/qpainterpath.cpp
-
-    ## All values are big endian--pack using struct.pack('>d') or struct.pack('>i')
+    # Create all vertices in path. The method used below creates a binary format so that all
+    # vertices can be read in at once. This binary format may change in future versions of Qt,
 
-    path = QtGui.QPainterPath()
+    # Speed this up using >> operator
+    # Format is:
+    #    numVerts(i4)
+    #    0(i4)   x(f8)   y(f8)    <-- 0 means this vertex does not connect
+    #    1(i4)   x(f8)   y(f8)    <-- 1 means this vertex connects to the previous vertex
+    #    ...
+    #    cStart(i4)   fillRule(i4)
+    #
+    # see: https://github.com/qt/qtbase/blob/dev/src/gui/painting/qpainterpath.cpp
 
+    # All values are big endian--pack using struct.pack('>d') or struct.pack('>i')
+    path = QtGui.QPainterPath()
     n = x.shape[0]
 
-    # create empty array, pad with extra space on either end
-    arr = np.empty(n+2, dtype=[('c', '>i4'), ('x', '>f8'), ('y', '>f8')])
-
-    # write first two integers
-    byteview = arr.view(dtype=np.ubyte)
-    byteview[:16] = 0
-    byteview.data[16:20] = struct.pack('>i', n)
+    backstore = bytearray(4 + n*20 + 8)
+    arr = np.frombuffer(backstore, dtype=[('c', '>i4'), ('x', '>f8'), ('y', '>f8')],
+        count=n, offset=4)
+    struct.pack_into('>i', backstore, 0, n)
 
     # Fill array with vertex values
-    arr[1:-1]['x'] = x
-    arr[1:-1]['y'] = y
+    arr['x'] = x
+    arr['y'] = y
+
 
     # inf/nans completely prevent the plot from being displayed starting on 
     # Qt version 5.12.3; these must now be manually cleaned out.
     isfinite = None
     qtver = [int(x) for x in QtVersion.split('.')]
-    if qtver >= [5, 12, 3]:
+    qt6 = qtver[0] == "6"
```

Comment:
Should be
```python
qt6 = qtver[0] == 6
```

Reply:
well, this one is embarrassing.  Thanks for pointing this out.